[Content] Remnant Teciimach Torpedo (#6707)
endless-sky/endless-sky: data/remnant/remnant outfits.txt

	category "Ammunition"
	licenses
		Remnant
	cost 35500
	thumbnail "outfit/emp torpedo"
	"mass" 1
	"teciimach canister capacity" -1
	description "This canister contains everything required to assemble a Teciimach torpedo. These weapons emit strong electromagnetic pulses which interfere with a ship's generators and neutralize a portion of their energy."
	description `	Unlike the previous generation of "EMP torpedoes," the Teciimach torpedoes actively shape their emission pattern to avoid impacting their allies.`

outfit "Teciimach Canister Rack"
	category "Ammunition"
	licenses
		Remnant
	cost 298000
	thumbnail "outfit/emp rack"
	"mass" 5
	"outfit space" -9
	"energy capacity" 500
	"teciimach canister capacity" 9
	ammo "Teciimach Canister"
	description "The Teciimach Canister rack is used to store extra sets of components, both to expand the capacity of a ship, and on carriers for resupplying fighters."
	description "	In addition to its ammunition capacity, the storage bay can also contain a small amount of charge, enough to let an otherwise battery-less ship assemble Teciimach torpedoes."

outfit "Teciimach Bay"
	category "Secondary Weapons"
	licenses
		Remnant

> 	cost 5250000

Reviewer: I'm happy with the ammo cost, but I actually think the bay cost is way too high now. For comparison, the Finisher Pod, which has an ammo cost that is now similar to the Teciimach (32k vs 35.5k), only has a bay cost of 1.37 mil compared to the Teciimach's 5.25 mil.

A bay cost in the 900k to 1.2 mil range feels more appropriate to me, and the pod cost could be roughly half that.

Author: Sure, that sounds fine. I'll make the change.

Author: Change has been made. Thanks for the suggestions!

		"missile strength" 80
	description "Electromagnetic pulse weapons were developed by the Remnant during the early days of their colony, when they were living in fear that the Alphas would overrun human space and expand beyond it."

outfit "Teciimach Canister"

> 	plural "Teciimach Canisters"

Reviewer: ```suggestion
```

	category "Ammunition"
	licenses
		Remnant
	cost 5500
	thumbnail "outfit/emp torpedo"
	"mass" 1
	"teciimach canister capacity" -1
	description "This canister contains everything required to assemble a Teciimach torpedo. These weapons emit strong electromagnetic pulses which interfere with a ship's generators and neutralize a portion of their energy."

> 	description "Unlike the previous generation of 'EMP torpedoes', the Teciimach torpedoes synchronize themselves with their mothership in order to avoid impacting their allies."

Reviewer: ```suggestion
	description `	Unlike the previous generation of "EMP torpedoes," the Teciimach torpedoes synchronize themselves with their mothership in order to avoid impacting their allies.`
```

outfit "Teciimach Canister Rack"

> 	plural "Teciimach Canister Racks"

Reviewer: ```suggestion
```

	category "Ammunition"
	licenses
		Remnant
	cost 98000
	thumbnail "outfit/emp rack"
	"mass" 5
	"outfit space" -9
	"energy capacity" 500

> 	"teciimach canister capacity" 9

Reviewer: 5 mass and 9 outfit space would imply a capacity of 4.
```suggestion
	"teciimach canister capacity" 4
```

Author: No, the masses are not cumulative with outfit space. This is deliberate.

	ammo "Teciimach Canister"
	description "The Teciimach Canister rack is used to store extra sets of components, both to expand the capacity of a ship, and on carriers for resupplying fighters."
	description "	In addition to its ammunition capacity, the storage bay can also contain a small amount of charge, enough to let an otherwise battery-less ship assemble Teciimach torpedoes."

outfit "Teciimach Bay"

> 	plural "Teciimach Bays"

Reviewer: No custom plural necessary.
```suggestion
```

	category "Secondary Weapons"
	licenses
		Remnant
	cost 583000
	thumbnail "outfit/teciimach bay"
	"mass" 9
	"outfit space" -13

> 	"weapon capacity" -13

Reviewer: 13 space and 9 ammo would imply a mass of 4. But I'll note how the Teciimach bay deals about 80% more energy drain (taking ionization and energy damages into account together), 30% more shield damage, and 25% more hull damage, and has 25% better turning, about 12% or 14% better tracking (depending on the tracking type), 19% better missile strength, and being a "safe" weapon, all while being about 70% the size, having the exact same bay cost and capacity, and only having the ammunition cost 25% more.

I would expect a weapon with this many additional benefits to either have a much larger and more expensive launcher with ammunition that's a bit more expensive than just +25%, or if it is to be made smaller then for it to have a reduced capacity, say 6 capacity with a mass of 7, with much more expensive ammunition, say 25k at the very least, but I'd be tempted to go above even that. (Note that even at a capacity of 6, a Teciimach would still output more total ion+energy damage than a normal EMP bay by about 18% in about 60% of the time.)

Author: Thank  you for pointing out the prices - I'd totally neglected to change those. I have fixed that now.

In terms of explanation, the prices are fairly massive increases. The main launcher is almost a tenfold increase in cost, and the ammo is a five fold increase. The pod is about half the price of the launcher. At this point in the Remnant story, the player has complete a fairly long string of missions that pay decently; and have access to jobs that pay quite well. They also have extremely easy access to farmable/lootable enemy ships that are an easy source of millions of credits. 

So on a meta level, this is both a balance to the easy loot economy that players at this stage have access to.

On the lore side, this has the triple hit of:
- being cutting edge tech that is upgrading their centuries old stuff.
- being a smaller economy that doesn't have the advanges of the massive mass production that, say, the Wanderers or Coalition can do, or even the Korath.
- being brand new, and thus higher prices of stuff just reaching distribution.

	"gun ports" -1
	"teciimach canister capacity" 9
	weapon
		sprite "projectile/emp torpedo"
			"frame rate" 2
			"no repeat"
		sound "torpedo"
		ammo "Teciimach Canister"
		icon "icon/emp"
		"fire effect" "emp torpedo fire" 5
		"hit effect" "nuke explosion"
		"hit effect" "emp ring" 30
		"hit effect" "emp spark" 20
		"die effect" "tiny explosion"
		cluster

> 		safe

Author: > "Well why doesn't anyone else?"

Primary answer would be "because no-one else uses EMPs." And even if they did, this would require careful synchronization that requires it to be connected somehow. Ex. If some Remnant ships went rogue and attacked other Remnant, they would be subject to each-other's EMPs, since they aren't sharing the deployment info required for this synchronization. The wave-cancelling we see in present day is just one possibility. There's a *lot* of other ways to mitigate electrical damage; and Star Trek is filled with techno-babble incidents of people synchronizing shields to effects in order to save the day; typically either an effect they control, or one they've had the chance to study at length.

> Perhaps the Remnant create an outfit alongside this weapon which provides a great deal of ion resistance? Basically a shield generator but for fighting ionization. All Remnant craft which are expected to be fielded alongside a Teciimach would then be equipped with this device, and even some ships that aren't expecting to be fighting alongside a Teciimach would likely use this device due to it helping with resisting ion storms.

I agree that the justification makes perfect sense, but I'd just like to point out that them having an outfit that can largely negate the effects of an EMP hit seems quite a bit more advanced / high-tech than simply having developed internal mechanisms for mitigating the effects of a weapon that they launched in the first place. This is why I went with the `safe` attribute instead of resistances. If you're good with resistances, though, I can do that.

I would opt for a distributed approach, though: 
- Some ionization protection on the ships themselves, fitting in with the lore of semi-organic ships that don't rely on classic electronics as much. They aren't any better than other ships at *removing* ionization, they just don't get as much of it to begin with. This would be scaled, so older (more electrical) ships would have less protection than newer ships.
- Buffed ionization resistance on the existing outfits that already have it. The resistance values are, so far as I can tell, too low for anyone to consider them as anything other than a slightly appreciated footnote.

> 		safe

Reviewer: > I agree that the justification makes perfect sense, but I'd just like to point out that them having an outfit that can largely negate the effects of an EMP hit seems quite a bit more advanced / high-tech than simply having developed internal mechanisms for mitigating the effects of a weapon that they launched in the first place.

I don't see how partially negating an EMP hit any more advanced than COMPLETELY negating one. Even if the explanation is something to the effect of "They do a lot with a little," gameplay wise a complete negation of something is more advanced than a partial negation of it.

> Some ionization protection on the ships themselves...

I'll give you 0.1 at the *most*. But really, given the numbers below, I'd rather not even bother with ion protection.

> Buffed ionization resistance on the existing outfits that already have it.

Ships vs the ionization of the Teciimach (65 ion), time for ionization to decay to various values / total energy lost by that point when equipped with nothing or one of the ion resistance Remnant outfits (Crystal Capacitor, Millennium Cell, Epoch Cell, Aeon Cell).

Ion value | 0 resistance | 0.01 resistance | 0.025 resistance | 0.085 resistance | 0.14 resistance
---|---|---|---|---|---
60 | 8 frames / 502 energy | 8 frames / 502 energy | 8 frames / 502 energy | 8 frames / 500 energy | 7 frames / 439 energy
50 | 27 frames / 1545 energy | 26 frames / 1492 energy | 26 frames / 1487 energy | 23 frames / 1321 energy | 21 frames / 1209 energy
40 | 49 frames / 2528 energy | 48 frames / 2478 energy | 47 frames / 2424 energy | 42 frames / 2174 energy | 38 frames / 1976 energy
30 | 77 frames / 3502 energy | 76 frames / 3449 energy | 73 frames / 3327 energy | 65 frames / 2973 energy | 59 frames / 2708 energy
20 | 118 frames / 4515 energy | 114 frames / 4387 energy | 110 frames / 4240 energy | 95 frames / 3714 energy | 84 frames / 3328 energy
10 | 187 frames / 5508 energy | 179 frames / 5329 energy | 168 frames / 5083 energy | 138 frames / 4341 energy | 119 frames / 3845 energy
5 | 256 frames / 6004 energy | 239 frames / 5764 energy | 219 frames / 5455 energy | 169 frames / 4569 energy | 142 frames / 4016 energy
1 | 416 frames / 6400 energy | 348 frames / 6052 energy | 295 frames / 5664 energy | 204 frames / 4670 energy | 166 frames / 4086 energy

And here's a table of the final results with comparisons against no resistance. Only taking the decay time and energy lost to get to an ionization of 1 since the last bit of ion takes a while to decay off, and gameplay wise isn't hurting you much. 

I've also included a rough equivalent amount of ion protection one would need to receive the same reduction in energy loss for the sake of comparing protection to resistance. Note that the equivalent protection for any level of resistance depends on the amount of ionization that was applied; resistance is actually better against smaller values of ionization (such as those received during an ion storm), whereas protection is the same flat buff to energy loss across all ionization values. Resistance being better at low values can be observed by how at different resistance levels, the amount of energy lost near the start is very close, then as time goes on the resistances deviate from one another in total energy lost.

Resistance | Decay time (frames) | % better than 0 | Energy lost | % better than 0 | Protection equivalent
---|---|---|---|---|---
0 | 416 | | 6400 | | 0
0.01 | 348 | 16.3% | 6052 | 5.4% | 0.057
0.025 | 295 | 29.1% | 5664 | 11.5% | 0.130
0.085 | 204 | 51.0% | 4670 | 27.0% | 0.370
0.14 | 166 | 60.1% | 4086 | 36.2% |  0.567

I compare both decay times and energy lost because the new effect of ionization jamming weapons means that decay time becomes just as if not more important than total energy lost, as decaying ionization quicker means getting jammed less.

Basically, I don't think any existing outfits need buffed. If a Starling with an Epoch cell or an Albatross with an Aeon Cell can recover more than twice as fast as whatever Korath ship it's attacking, that's pretty good already.

> Having a specific anti-ionization outfit would be a separate unlock requiring more missions, so would be a separate PR for a later date.

I'm fine with that if we don't use `safe` here. Given the numbers above, I would say that an ion resistance of 0.03 on a relatively small dedicated outfit (say, 17 tons) might work. That's roughly double the resistance per ton of the existing Remnant outfits. Although unlike the passive ion resistance of those outfits, I'd say giving this dedicated outfit an energy cost would make sense, say `"ion resistance energy" 3`.

> 		safe

Reviewer 2: "ion resistance energy" of 3 on an "ion resistance" of 0.03 essentially means it'll remove all the energy the 0.03 ionisation would have in its lifetime immediately. That's not super useful. It removes the same energy over a much shorter time (which is arguably worse). It does have the benefit of reducing the amount of weapon jams the ship is likely to experience but that's it. Doubling the resistance per ton of the outfits that are also power generators or storage doesn't seem like enough to me. How would installing another crystal capacitor compare to this outfit in terms of actual performance against ion storms and various ionising weapons?

> 		safe

Author: The other problem would be that no amount of ionization resistance or protection will do anything for the rest of the Teciimach's effects. The Shield damage and energy hit are just as big a problem as the ionization.

> 		safe

Reviewer: > It does have the benefit of reducing the amount of weapon jams the ship is likely to experience but that's it.

That's the intention. Preventing jamming can be a big boost, so I wouldn't say that it's "not super useful."

> How would installing another crystal capacitor compare to this outfit in terms of actual performance against ion storms and various ionising weapons?

Well crystal capacitors already consume 0.4 energy per frame, so you could perhaps imagine that as crystal capacitors having roughly half the ion resistance per ton in exchange for being roughly twice as energy efficient with bleeding ionization. Although crystal capacitors constantly drain energy, not just when bleeding ionization, so perhaps whether the reduced energy drain at a constant rate is better or worse is up to the user.

> 		safe

Author: > I don't see how partially negating an EMP hit is any more advanced than COMPLETELY negating one. Even if the explanation is something to the effect of "They do a lot with a little," gameplay wise a complete negation of something is more advanced than a partial negation of it.

Because it is selectively negating *only* the effects of a a detonation where the launching ship knows, to the nanosecond, the exact time of launch, time of detonation, exact characteristics of said detonation in every respect, etc. Any other circumstance, such as the same weapon launched by a hostile ship? Zero protection. A sort of similar weapon from another faction? Also zero protection. 

So swapping the "safe" function here for some very minor ionization resistance is trading the ability to use this weapon effectively for protection against *every* weapon that makes use of ionization, no matter the source, mechanism, or type.

----

Since it seems to be the capability to synchronize shields that is the problem, would changing the description make it more palatable? For instance, stating that the weapon burst has a mechanism to control the directionality of its emissions to only affect enemy ships? I'd be fine with that.

> 		safe

Reviewer 2: > And, again, I'd like to point out that everyone is focusing really heavily on the ionization part, and ignoring that this weapon actually does do shield damage, energy damage, and hull damage. No amount of ionization resistances are going to mitigate those other aspects.

I don't think that actually helps your argument that allied ships being able to prepare for the blast enables the `safe` functionality.

> to the nanosecond, the exact time of launch, time of detonation, exact characteristics of said detonation in every respect

That's a lot of very precise details. I have trouble believing that even the firing ship would know when the projectile was launched *to the nanosecond*, let alone knowing when it will detonate to such precision. And implies a high level of predictability to these missiles. If "shield polarisation" can negate all of these things so completely because so much is known about the detonation, why should it not be possible to adapt to incoming enemy fire with sufficient analysis?

If the shielding is the justification for the `safe`ness why does it persist even when shields are fully drained? What if I have a ship with no shields?

> Since it seems to be the capability to synchronize shields that is the problem, would changing the description make it more palatable? For instance, stating that the weapon burst has a mechanism to control the directionality of its emissions to only affect enemy ships?

Yes, please.

> 		safe

Author: Note: The mechanism has been changed, so it is now an actively shaped emission pattern rather than synchronization with the launching ship. See recent commits for specifics.

> 		safe

Reviewer 3: Quick note, EMP can scale upwards, so unless you've protected the ship in question to withstand anything higher than a Lightning Strike, it's unlikely your vessel could withstand a dedicated EMP weapon.

That, and inactive electronics aren't as badly affected as active ones.

> 		safe

Author: I think this aspect as been more or less resolved, so marking it as such for clarity.

		"reload" 310
		"firing energy" 400
		"firing heat" 18
		"inaccuracy" 5
		"lifetime" 300
		"velocity" 8
		"acceleration" .7
		"drag" .1
		"turn" 2.5
		"homing" 3
		"infrared tracking" .8
		"radar tracking" .9
		"trigger radius" 50
		"blast radius" 200
		"shield damage" 900
		"hull damage" 125

> 		"disabled damage" 60

Author: Hence why it still does disabled damage instead of being zeroed out entirely. Also, my assumption is that a disabled ship does have physical problems. If the wire isn't even connected to the computer anymore because a chunk got blasted out of it earlier in the battle, then that wire isn't going to be conducting charge into the computer. I'm not saying that there's nothing left to affect, I'm saying that the pathways to concentrate charge at dangerous levels are going to be vastly reduced. If you have a ham radio that is powered on and hooked up to an antenna, it will take a lot more EMP damage than an equivalent radio that is unpowered and disconnected from the antenna.

Also, what other EMP weapons? There's the Ka'het (which might be an inspiration for the Remnant...), but I don't see any others. I don't see any balance issues with a non-purchasable thing like the Ka'het emp deployer have a similar disable damage. It only does 100 hull damage (compared to 700 shield damage), so having it drop to 60 disabled damage as well is not a huge drop.

> 		"disabled damage" 60

Reviewer: >  If the wire isn't even connected to the computer anymore because a chunk got blasted out of it earlier in the battle, then that wire isn't going to be conducting charge into the computer.

But the computer itself contains wires. It doesn't matter if it's connected to something else or not, it is the wires within the electronics that get overloaded and shorted out by an EMP.

> If you have a ham radio that is powered on and hooked up to an antenna, it will take a lot more EMP damage than an equivalent radio that is unpowered and disconnected from the antenna.

Are you sure? Because everything I see says that powering off electronics is no protection against an EMP.

> 		"disabled damage" 60

Author: As best I can tell, it does. The powering off itself may not protect it; but disconnecting it from things that can pick up and channel more of the surge into it *can*. Also, circuits that are off can (and often will be, in the case of a ship that's taken a lot of damage) splintered into separate parts so that whereas the EMP hitting a wire in, say, the nose of the ship, won't be conducted to a computer in the engine room. 

As such, once the ship is disabled, each component only needs to deal with the EMP that is hitting it in specific, rather than being subject to all the conflicting waves of energy being transmitted through the wiring, which could effectivley lead to components being subject to energy spikes multiple times.

There's also the fact that some of that damage may have shorted and/or grounded out wiring, which further isolates components from additional damage. (proper grounding is one method of minimizing damage to electronics, too; but it is very hard to ground out everything while its running.)

> 		"disabled damage" 60

Reviewer: I think we're missing the point that hull damage is only done if the shields are down. If the argument being made is that the ionization overloads electronics and causes hull damage, then I would expect that hull damage to be dealt regardless of whether the shields are up or not, perhaps simply with a reduced impact against shielded targets just as the ion damage is shielded. That implies piercing, but I'd rather not tack even more onto weapons such as this. I would also expect all ion weapons to have this behavior, but they don't, and I don't think we should give that behavior to them.

> 		"disabled damage" 60

Author: I will just point out that while EMPs may ionize, not all ionization is an EMP. These are two distinct effects that just happen to include some of the same effects, and, which I'm trying to move a little farther apart. The Teciimach is *not* just a really high power ion canon.

> 		"disabled damage" 60

Author: To expand on that, I view ionization as a comparatively gentle charging of the whole ship to cause negative interactions with ship systems. EMP is an instantaneous burst intended to overload electrical circuitry, with the emphasis on being a hard "impact" to anything that relies on conductive systems. In this case, the ionization is more of an after-effect of the primary effect.

		"ion damage" 65
		"energy damage" 2000
		"hit force" 90
		"missile strength" 95
	description "Electromagnetic pulse weapons were originally developed by the Remnant during the early days of their colony, when they were living in fear that the Alphas would overrun human space and expand beyond it."
	description "Centuries later, the Teciimach revision includes a range of upgrades to their effectiveness, efficiency, reliability, and safety. The Remnant attribute these improvements to a mix of the latest research and secrets uncovered by Warren research teams."

outfit "Teciimach Pod"
	category "Secondary Weapons"
	licenses
		Remnant

> 	cost 583000

Reviewer: Same cost as the bay?

Author: Good spotting. Will change it.

Author: Has now been changed. Thanks for pointing that out.

	thumbnail "outfit/teciimach pod"

> 	"mass" 5

Reviewer: 6 outfit space and 3 ammo implies 3 mass.
```suggestion
	"mass" 3
```

	"outfit space" -6
	"weapon capacity" -6
	"gun ports" -1
	"teciimach canister capacity" 3
	weapon
		sprite "projectile/emp torpedo"
			"frame rate" 2
			"no repeat"
		sound "torpedo"
		ammo "Teciimach Canister"
		icon "icon/emp"
		"fire effect" "emp torpedo fire" 5
		"hit effect" "nuke explosion"
		"hit effect" "emp ring" 30
		"hit effect" "emp spark" 20
		"die effect" "tiny explosion"
		cluster
		safe
		"burst count" 3
		"burst reload" 110
		"reload" 620
		"firing energy" 350
		"firing heat" 16
		"inaccuracy" 5
		"lifetime" 300
		"velocity" 8
		"acceleration" .7
		"drag" .1
		"turn" 2.5
		"homing" 3
		"infrared tracking" .8
		"radar tracking" .9
		"trigger radius" 50
		"blast radius" 200
		"shield damage" 900
		"hull damage" 125
		"disabled damage" 60
		"ion damage" 60
		"energy damage" 2000
		"hit force" 90
		"missile strength" 95

> 	description "While originally developed as a heavy weapon to paralyze enemy capital ships, the Remnant have utilized on-the-spot assembly techniques to support their use on smaller ships. These pods sacrifice multiple assembly systems in order to fit them onto interceptor class vessels."

Reviewer 4: ```suggestion
	description "While originally developed as a heavy weapon to paralyze enemy capital ships, the Remnant have utilized on-the-spot assembly techniques to support their use on smaller ships. These pods sacrifice multiple assembly systems in order to fit them onto interceptor-class vessels."
```

> 	description "Despite this explanation, a careful examination of these bays and pods suggest that there is much more here than their volume would suggest."

Reviewer: ```suggestion
	description "	Despite this explanation, a careful examination of these bays and pods suggest that there is much more here than their volume would suggest."
```
